step/build_id_getter.go

```diff
@@ -68,6 +68,9 @@ func (bg BuildIDGetter) createKeyValuePairSlice() []keyValuePair {
 	var stageWorkflowMap []keyValuePair
 	for _, stage := range bg.FinishedStages {
 		for _, wf := range stage.Workflows {
+			if wf.ExternalId == "" {
```

Comment:
shouldn't we log this?

Reply:
Plus one on this one. I strongly prefer to log something to the customer so they can see in the logs what's going on.